### ISSUE TITLE
estimate_gas failure should not close connection with master

### ISSUE THREAD

**Issue by qcgg** [QuarkChain/pyquarkchain#155]
```
MASTER: I1012 04:33:34.078363 jsonrpc.py:408] {"jsonrpc":"2.0","method":"estimateGas","id":"1","params":{"data":"0xa6f2ae3a","to":"0x837f9A0e7185141D75e37D55e213efd725e76710B10d0E9F","value":"0xde0b6b3a7640000","from":"0xb0997B1309C61D50495b79b2B82D377d862Db367b0c6797f"}}
SLAVE_S31: ESC[1;31mE1012ESC[1;0m 04:33:34.135198 protocol.py:167] Traceback (most recent call last):
SLAVE_S31: File "/code/pyquarkchain/quarkchain/protocol.py", line 165, in __internal_handle_metadata_and_raw_data
SLAVE_S31: await self.handle_metadata_and_raw_data(metadata, raw_data)
SLAVE_S31: File "/code/pyquarkchain/quarkchain/cluster/protocol.py", line 80, in handle_metadata_and_raw_data
SLAVE_S31: await super().handle_metadata_and_raw_data(metadata, raw_data)
SLAVE_S31: File "/code/pyquarkchain/quarkchain/protocol.py", line 152, in handle_metadata_and_raw_data
SLAVE_S31: await self.__handle_rpc_request(op, cmd, rpc_id, metadata)
SLAVE_S31: File "/code/pyquarkchain/quarkchain/protocol.py", line 127, in __handle_rpc_request
SLAVE_S31: resp = await handler(self, request)
SLAVE_S31: File "slave.py", line 430, in handle_estimate_gas
SLAVE_S31: res = self.slave_server.estimate_gas(req.tx, req.from_address)
SLAVE_S31: File "slave.py", line 1136, in estimate_gas
SLAVE_S31: return shard.state.estimate_gas(tx, from_address)
SLAVE_S31: File "/code/pyquarkchain/quarkchain/cluster/shard_state.py", line 1290, in estimate_gas
SLAVE_S31: if hi == cap and not run_tx(hi):
SLAVE_S31: File "/code/pyquarkchain/quarkchain/cluster/shard_state.py", line 1277, in run_tx
SLAVE_S31: evm_tx = self.__validate_tx(tx, evm_state, from_address, gas=gas)
SLAVE_S31: File "/code/pyquarkchain/quarkchain/cluster/shard_state.py", line 255, in __validate_tx
SLAVE_S31: validate_transaction(evm_state, evm_tx)
SLAVE_S31: File "/code/pyquarkchain/quarkchain/evm/messages.py", line 178, in validate_transaction
SLAVE_S31: rp(tx, "balance", state.get_balance(tx.sender), total_cost)
SLAVE_S31: quarkchain.evm.exceptions.InsufficientBalance: <Transaction(09e6)>: 'balance' actual:8858050000000000 target:1000000000000000000
SLAVE_S31: 
SLAVE_S31: I1012 04:33:34.135638 slave.py:148] Closing connection with master: slave_master: error processing request: <Transaction(09e6)>: 'balance' actual:8858050000000000 target:1000000000000000000
SLAVE_S31: I1012 04:33:34.135953 slave.py:144] Lost connection with master
SLAVE_S31: I1012 04:33:34.136633 slave.py:144] Lost connection with master
SLAVE_S31: I1012 04:33:34.136918 slave.py:144] Lost connection with master
```

**Comment by ninjaahhh** [comment]
reopen the issue because of [this discussion](https://github.com/QuarkChain/pyquarkchain/pull/156#discussion_r224678549)

**Comment by ninjaahhh** [comment]
closed by #160
